pkg/configurator/client.go

```diff
@@ -135,67 +156,72 @@ func (c *Client) getConfigMap() *osmConfig {
 
 	configMap := item.(*v1.ConfigMap)
 
-	osmConfigMap := osmConfig{
-		PermissiveTrafficPolicyMode: getBoolValueForKey(configMap, permissiveTrafficPolicyModeKey),
-		Egress:                      getBoolValueForKey(configMap, egressKey),
-		EnableDebugServer:           getBoolValueForKey(configMap, enableDebugServer),
-		PrometheusScraping:          getBoolValueForKey(configMap, prometheusScrapingKey),
-		UseHTTPSIngress:             getBoolValueForKey(configMap, useHTTPSIngressKey),
-
-		TracingEnable:               getBoolValueForKey(configMap, tracingEnableKey),
-		EnvoyLogLevel:               getStringValueForKey(configMap, envoyLogLevel),
-		ServiceCertValidityDuration: getStringValueForKey(configMap, serviceCertValidityDurationKey),
-	}
+	// Parse osm-config ConfigMap.
+	// In case of missing/invalid value for a key, osm-controller uses the default value.
+	// Invalid values should be prevented once https://github.com/openservicemesh/osm/issues/1788
+	// is implemented.
+	osmConfigMap := osmConfig{}
+	osmConfigMap.PermissiveTrafficPolicyMode, _ = GetBoolValueForKey(configMap, PermissiveTrafficPolicyModeKey)
+	osmConfigMap.Egress, _ = GetBoolValueForKey(configMap, egressKey)
+	osmConfigMap.EnableDebugServer, _ = GetBoolValueForKey(configMap, enableDebugServer)
+	osmConfigMap.PrometheusScraping, _ = GetBoolValueForKey(configMap, prometheusScrapingKey)
+	osmConfigMap.UseHTTPSIngress, _ = GetBoolValueForKey(configMap, useHTTPSIngressKey)
+	osmConfigMap.TracingEnable, _ = GetBoolValueForKey(configMap, tracingEnableKey)
+	osmConfigMap.EnvoyLogLevel, _ = GetStringValueForKey(configMap, envoyLogLevel)
+	osmConfigMap.ServiceCertValidityDuration, _ = GetStringValueForKey(configMap, serviceCertValidityDurationKey)
 
 	if osmConfigMap.TracingEnable {
-		osmConfigMap.TracingAddress = getStringValueForKey(configMap, tracingAddressKey)
-		osmConfigMap.TracingPort = getIntValueForKey(configMap, tracingPortKey)
-		osmConfigMap.TracingEndpoint = getStringValueForKey(configMap, tracingEndpointKey)
+		osmConfigMap.TracingAddress, _ = GetStringValueForKey(configMap, tracingAddressKey)
+		osmConfigMap.TracingPort, _ = GetIntValueForKey(configMap, tracingPortKey)
+		osmConfigMap.TracingEndpoint, _ = GetStringValueForKey(configMap, tracingEndpointKey)
```

Comment:
A little worried about the `_` here - at some point we should check for errors coming from the gets

Reply:
I agree, that's the existing behavior though. I think a validating webhook for osm-config would prevent such problems. Historically, the reason the error is ignored is also because some of the ConfigMap keys are optional with reasonable defaults, so if a key is missing in the ConfigMap, should the controller error? We need to establish the contract between Helm chart ConfigMap creation and osm-controller expectations.